extensions/azure/vault/src/main/java/org/eclipse/dataspaceconnector/security/azure/AzureVaultExtension.java

```diff
@@ -55,28 +55,31 @@ public CertificateResolver getCertificateResolver() {
     }
 
     @Override
-    public void intializeVault(ServiceExtensionContext context) {
+    public void initializeVault(ServiceExtensionContext context) {
         String clientId = context.getSetting("edc.vault.clientid", null);
         if (isNullOrEmpty(clientId)) {
-            throw new AzureVaultException("'edc.vault.clientid' must be supplied but was null!");
+            throw new AzureVaultException("'edc.vault.clientid' must be supplied but was null");
         }
 
         String tenantId = context.getSetting("edc.vault.tenantid", null);
         if (isNullOrEmpty(tenantId)) {
-            throw new AzureVaultException("'edc.vault.tenantid' must be supplied but was null!");
+            throw new AzureVaultException("'edc.vault.tenantid' must be supplied but was null");
         }
 
+        String clientSecret = context.getSetting("edc.vault.secret", null);
```

Comment:
As this introduces a bit of logic, I think a unit test is necessary. This should also verify the precedence: when both the cert path and a secret are specified, the cert path takes precedence over the secret.

Reply:
Done!